module/proto/gravity/v1/genesis.proto

```diff
@@ -72,47 +70,56 @@ option go_package = "github.com/cosmos/gravity-bridge/module/x/gravity/types";
+  uint64 bridge_chain_id                 = 5;
+  uint64 signed_valsets_window           = 6;
+  uint64 signed_batches_window           = 7;
+  uint64 signed_claims_window            = 8;
+  uint64 target_batch_timeout            = 10;
+  uint64 average_block_time              = 11;
+  uint64 average_ethereum_block_time     = 12;
+  bytes  slash_fraction_valset           = 13 [
     (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
     (gogoproto.nullable)   = false
   ];
-  bytes slash_fraction_batch = 14 [
+  bytes slash_fraction_batch             = 14 [
     (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
     (gogoproto.nullable)   = false
   ];
-  bytes slash_fraction_claim = 15 [
+  bytes slash_fraction_claim             = 15 [
     (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
     (gogoproto.nullable)   = false
   ];
   bytes slash_fraction_conflicting_claim = 16 [
     (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
     (gogoproto.nullable)   = false
   ];
-  uint64 unbond_slashing_valsets_window = 17;
+  uint64 unbond_slashing_valsets_window  = 17;
 }
 
 // GenesisState struct
+// TODO: this need to be audited and potentially simplified using the new
+// interfaces
 message GenesisState {
-  Params                             params              = 1;
-  uint64                             last_observed_nonce = 2;
-  repeated Valset                    valsets             = 3;
-  repeated MsgValsetConfirm          valset_confirms     = 4;
-  repeated OutgoingTxBatch           batches             = 5;
-  repeated MsgConfirmBatch           batch_confirms      = 6 [(gogoproto.nullable) = false];
-  repeated OutgoingLogicCall         logic_calls         = 7;
-  repeated MsgConfirmLogicCall       logic_call_confirms = 8 [(gogoproto.nullable) = false];
-  repeated Attestation               attestations        = 9 [(gogoproto.nullable) = false];
-  repeated MsgSetOrchestratorAddress delegate_keys       = 10;
-  repeated ERC20ToDenom              erc20_to_denoms     = 11;
-  repeated OutgoingTransferTx        unbatched_transfers = 12;
+  Params                              params                         = 1;
+  uint64                              last_observed_nonce            = 2;
+  repeated UpdateSignerSetTx          valsets                        = 3;
+  repeated UpdateSignerSetTxSignature valset_confirms                = 4;
+  repeated BatchTx                    batches                        = 5;
+  repeated BatchTxSignature           batch_confirms                 = 6 [(gogoproto.nullable) = false];
+  repeated ContractCallTx             logic_calls                    = 7;
+  repeated ContractCallTxSignature    logic_call_confirms            = 8 [(gogoproto.nullable) = false];
+  repeated EthereumEventVoteRecord    ethereum_event_vote_records    = 9 [(gogoproto.nullable) = false];
+  repeated MsgDelegateKeys            delegate_keys                  = 10;
```

Comment:
Why is this the only thing here that still has the Msg prefix? Was it forgotten?

Reply:
Because we aren't persisting things in state as their `Msg` struct in most places anymore. In this case, the `MsgDelegateKeys` struct is a good way to persist this data.